caravel/assets/visualizations/nvd3_vis.js

```diff
@@ -295,8 +312,9 @@ function nvd3Vis(slice) {
         }
 
         if (fd.bottom_margin) {
-          chart.margin({ bottom: fd.bottom_margin });
+          chart.margin({ bottom: Math.max(fd.bottom_margin, stretchMargin) });
```

Comment:
I feel like the precedence rule should not be Math.max but instead be the explicit (user defined) should win over implicit (calculated) when specified.